test/ssl_ctx_test.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "testutil.h"
+#include <openssl/ssl.h>
+
+typedef struct {
+    int min_version;
+    int max_version;
+    int min_ok;
+    int max_ok;
+    int expected_min;
+    int expected_max;
+} version_test;
+
+static const version_test version_testdata[] = {
+    /* min           max             ok    expected min    expected max */
+    {0,              0,              1, 1, 0,              0},
+    {TLS1_VERSION,   TLS1_2_VERSION, 1, 1, TLS1_VERSION,   TLS1_2_VERSION},
+    {TLS1_2_VERSION, TLS1_2_VERSION, 1, 1, TLS1_2_VERSION, TLS1_2_VERSION},
+    {TLS1_2_VERSION, TLS1_1_VERSION, 1, 0, TLS1_2_VERSION, 0},
```

Comment:
I'm not sure why this test passes. I expected setting the max version should in fact work, but fail when you try to connect using it.